src/ocl/batchnormocl.cpp

```diff
@@ -679,36 +679,36 @@ void BatchNormForwardInference(Handle& handle,
         unsigned int in_nstride = c * h * w;
         unsigned int in_cstride = h * w;
 
-        size_t xlocalsize = 1;
-        auto ylocalsize   = size_t((in_cstride > 1024) ? 1024 : ((64 >= in_cstride) ? 64 : 256));
-
-        std::vector<size_t> vld;
-        std::vector<size_t> vgd;
-
-        auto segment   = std::ceil(double(in_cstride) / double(ylocalsize));
-        auto xgridsize = size_t(c);
-        auto ygridsize = size_t(segment * ylocalsize);
-
-        std::string algo_name = "miopenBatchNormalizationForwardInference";
-        std::string network_config =
-            "n" + std::to_string(n) + +"c" + std::to_string(c) + "hw" + std::to_string(in_cstride) +
-            "chw" + std::to_string(in_nstride) + "segment" + std::to_string(segment) + "gx" +
-            std::to_string(xgridsize) + "gy" + std::to_string(ygridsize) + "lx" +
-            std::to_string(xlocalsize) + "ly" + std::to_string(ylocalsize) + "fp16" +
-            std::to_string(static_cast<int>(bfp16parm)) + "fp32" +
-            std::to_string(static_cast<int>(bfp32parm)) + "mode" + std::to_string(bn_mode);
+        std::string algo_name      = "miopenBatchNormalizationForwardInference";
+        std::string network_config = "fp16" + std::to_string(static_cast<int>(bfp16parm)) + "fp32" +
+                                     std::to_string(static_cast<int>(bfp32parm)) + "mode" +
+                                     std::to_string(bn_mode);
 
         auto&& kernels = handle.GetKernels(algo_name, network_config);
         if(!kernels.empty())
         {
             auto kernel = kernels.front();
-            kernel(x, y, estimatedMean, estimatedVariance, bnScale, bnBias, epsilon);
+            kernel(x,
+                   y,
+                   estimatedMean,
+                   estimatedVariance,
+                   bnScale,
+                   bnBias,
+                   epsilon,
+                   n,
+                   in_cstride,
+                   in_nstride,
+                   c);
         }
         else
         {
+            size_t xlocalsize = 1;
```

Comment:
Does these changes impact any performance?

Reply:
Performance is worse across the board. Attempting to lift it up a bit.

Reply:
Check the top level comment.